client/object_get.go

```diff
@@ -186,8 +185,7 @@ func (x *ObjectReader) readChunk(buf []byte) (int, bool) {
 			return read, false
 		}
 
-		_, x.err = x.client.processResponse(&resp)
-		if x.err != nil {
+		if x.err = x.client.processResponse(&resp); x.err != nil {
```

Comment:
I'd really prefer
```
x.err = x.client.processResponse(&resp)
if x.err != nil {
```
stylistically because it's a state change. It's kinda OK to have
```
if err := x.client.processResponse(&resp); err != nil {
```
when `err` is scoped, but `x.err` changed can be missed on a quick glance in this construction.

Not critical though.

Reply:
extracted from if scope